### ISSUE TITLE
React defaultProps are not working with targets: { node: true}

### ISSUE THREAD

**Issue by seeden** [babel/babel#7281]
Hi guys.

When I use 
```js
const babelQuery = {
  babelrc: false,
  plugins: [
    ['@babel/plugin-transform-proto-to-assign'],
    ['@babel/plugin-transform-flow-strip-types'],
  ],
  presets: [
    ['@babel/env', {
      targets: {
        node: true,
      },
      useBuiltIns: 'entry',
      loose: true,
    }],
    '@babel/stage-0',
    '@babel/react',
  ],
};
```

I can see next error.
We don't know what to do with this node type. We were previously a Statement but we can't fit in here?

```js
ERROR in ./src/components/base/Base.jsx
Module build failed: Error: We don't know what to do with this node type. We were previously a Statement but we can't fit in here?
    at NodePath.insertBefore (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/path/modification.js:44:11)
    at PluginPass.Class (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/plugin-proposal-class-properties/lib/index.js:225:14)
    at newFn (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/visitors.js:223:21)
    at NodePath._call (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/path/context.js:64:19)
    at NodePath.call (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/path/context.js:38:17)
    at NodePath.visit (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/path/context.js:99:12)
    at TraversalContext.visitQueue (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:139:18)
    at TraversalContext.visitSingle (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:98:19)
    at TraversalContext.visit (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:180:19)
    at Function.traverse.node (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/index.js:76:17)
    at NodePath.visit (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/path/context.js:106:18)
    at TraversalContext.visitQueue (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:139:18)
    at TraversalContext.visitMultiple (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:93:17)
    at TraversalContext.visit (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:178:19)
    at Function.traverse.node (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/index.js:76:17)
    at NodePath.visit (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/path/context.js:106:18)
    at TraversalContext.visitQueue (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:139:18)
    at TraversalContext.visitSingle (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:98:19)
    at TraversalContext.visit (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/context.js:180:19)
    at Function.traverse.node (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/index.js:76:17)
    at traverse (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/traverse/lib/index.js:46:12)
    at transformFile (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/core/lib/transformation/index.js:107:27)
    at runSync (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/core/lib/transformation/index.js:35:3)
    at transformSync (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/core/lib/transform-sync.js:15:38)
    at Object.transform (/Users/seeden/Documents/git/quizana-ui/node_modules/@babel/core/lib/transform.js:20:65)
    at transpile (/Users/seeden/Documents/git/quizana-ui/node_modules/babel-loader/lib/index.js:55:20)
    at Object.module.exports (/Users/seeden/Documents/git/quizana-ui/node_modules/babel-loader/lib/index.js:179:20)
 @ ./src/components/miss/MissPage.jsx 4:0-32
 @ ./src/components/Routes.jsx
 @ ./src/components/App.jsx
 @ ./src/index.js
```

It will desapear when I remove react defaultProps from my source code.

```js
static defaultProps = {};
```

It is working fine when I use browsers: ['> 1%', 'last 3 versions'] instead of node: true.

Here is my packages:
```js

    "babel-core": "7.0.0-bridge.0",
    "@babel/core": "^7.0.0-beta.38",
    "@babel/plugin-transform-proto-to-assign": "^7.0.0-beta.38",
    "@babel/plugin-transform-flow-strip-types": "^7.0.0-beta.38",
    "@babel/preset-env": "^7.0.0-beta.38",
    "@babel/preset-react": "^7.0.0-beta.38",
    "@babel/preset-stage-0": "^7.0.0-beta.38",
    "@babel/polyfill": "^7.0.0-beta.38",
    "@babel/register": "^7.0.0-beta.38",
    "babel-loader": "8.0.0-beta.0",
```

**Comment by Andarist** [comment]
@seeden Could you paste in faulty code? Preferably slimmed down to the minimal reproduction case.

**Comment by seeden** [comment]
Hi @Andarist. 
It is somehow connected with decorators.
```js
import React, { Component } from 'react';
import { withRouter } from 'react-router';

@withRouter
export default class TabsTab extends Component {
  static defaultProps = {
  };
}
```

It is working fine without decorator

**Comment by seeden** [comment]
It is working with 7.0.0-beta.36 but it is broken from 7.0.0-beta.36

**Comment by edsrzf** [comment]
I'm able to reproduce this with Babel `7.0.0-beta.38`, but it looks like this was fixed sometime before the 7.0.0 release. I cannot reproduce using `7.0.0` or the latest `7.6.2`. Would be nice to understand more details, but perhaps we can close this off?

**Comment by Andarist** [comment]
Seems like possible this got fixed along the way (as investigated by @edsrzf - thank you for that!). If the issue still persists please report it back (in a new issue), preferably with a repro case attached.

Closing this now as the issue seems stale.
